From 7228cf5e079a3b2d2441c994bcf90349d36239d8 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Wed, 20 Feb 2019 16:46:32 +0100
Subject: [PATCH 1040/1051] arm64: dts: cn9132: update serdes topology for
 setup A

Serdes configuration was update therefore currently supported devices
connected to phy on CP2 are:
- PCIex2 port0
- SATA port0
- USB3.0 port1
- 10G ETH port0
- PCIex1 port2

The SW1 found in the extension board "SLM-1521-V2" connected to CP2
should be change to:
SW1.1 = ON
SW1.2 = OFF
SW1.3 = ON
SW1.4 = OFF
SW1.5 = ON

It is required in order to connect appropriate serdes lane to relevant
devices supported in updated setup.

Change-Id: I0a6c7861e02c8e7c3275eac34c5508ffd0b91846
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/4699
Reviewed-by: Nadav Haklai <Nadav.Haklai@cavium.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Tested-by: sa_ip-sw-jenkins
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/cn9132-db-A.dts | 28 +++++++++++++++------
 1 file changed, 21 insertions(+), 7 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/cn9132-db-A.dts b/arch/arm64/boot/dts/marvell/cn9132-db-A.dts
index f82194329c20..139f9a49ebde 100644
--- a/arch/arm64/boot/dts/marvell/cn9132-db-A.dts
+++ b/arch/arm64/boot/dts/marvell/cn9132-db-A.dts
@@ -37,22 +37,29 @@
 /* SLM-1521-V2, CON6 */
 &cp2_pcie0 {
 	status = "okay";
-	num-lanes = <4>;
+	num-lanes = <2>;
 	num-viewport = <8>;
 	/* Generic PHY, providing serdes lanes */
 	phys = <&cp2_comphy0 0
-		&cp2_comphy1 0
-		&cp2_comphy2 0
-		&cp2_comphy3 0>;
+		&cp2_comphy1 0>;
+};
+
+/* SLM-1521-V2, CON8 */
+&cp2_pcie2 {
+	status = "okay";
+	num-lanes = <1>;
+	num-viewport = <8>;
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp2_comphy5 2>;
 };
 
 &cp2_sata0 {
 	status = "okay";
-	/* SLM-1521-V2, CON2 */
-	sata-port@1 {
+	/* SLM-1521-V2, CON4 */
+	sata-port@0 {
 		status = "okay";
 		/* Generic PHY, providing serdes lanes */
-		phys = <&cp2_comphy5 1>;
+		phys = <&cp2_comphy2 0>;
 	};
 };
 
@@ -60,3 +67,10 @@
 &cp2_sdhci0 {
 	status = "okay";
 };
+
+/* SLM-1521-V2, CON11 */
+&cp2_usb3_1 {
+	status = "okay";
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp2_comphy3 1>;
+};
-- 
2.17.1

